packages/caravan-fees/src/btcTransactionComponents.ts

```diff
@@ -0,0 +1,331 @@
+    return Boolean(
+      this._txid &&
+        this._vout >= 0 &&
+        (this._nonWitnessUtxo || this._witnessUtxo),
+    );
+  }
+
+  /**
+   * Converts the input template to a UTXO object.
+   */
+  toUTXO(): UTXO {
+    return {
+      txid: this._txid,
+      vout: this._vout,
+      value: this._amountSats.toString(),
+      prevTxHex: this._nonWitnessUtxo?.toString("hex"),
+      witnessUtxo: this._witnessUtxo,
+    };
+  }
+}
+/**
+ * Represents a Bitcoin transaction output
+ */
+export class BtcTxOutputTemplate extends BtcTxComponent {
+  private readonly _address: string;
+  private _malleable: boolean = true;
+
+  /**
+   * @param params - Output parameters
+   * @param params.address - Recipient address
+   * @param params.amountSats - Amount in satoshis  (as a string)
+   * @param params.locked - Whether the output is locked (immutable), defaults to false
+   * @throws Error if trying to create a locked output with zero amount
+   */
+  constructor(params: {
+    address: string;
+    amountSats?: Satoshis | undefined;
+    locked?: boolean;
+  }) {
+    super(params.amountSats || "0");
+    this._address = params.address;
+    this._malleable = !params.locked;
+
+    if (!this._malleable && this._amountSats.isEqualTo(0)) {
+      throw new Error("Locked outputs must have an amount specified.");
+    }
+  }
+
+  /** Get the recipient address */
+  get address(): string {
+    return this._address;
+  }
+
+  /** Check if the output is malleable (can be modified) */
+  get isMalleable(): boolean {
+    return this._malleable;
+  }
+
+  /**
+   * Set a new amount for the output
+   * @param amountSats - New amount in satoshis(as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  setAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = new BigNumber(amountSats);
+  }
+
+  /**
+   * Add amount to the output
+   * @param amountSats - Amount to add in satoshis (as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  addAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = this._amountSats.plus(new BigNumber(amountSats));
+  }
+
+  /**
+   * Subtract amount from the output
+   * @param amountSats - Amount to subtract in satoshis (as a string)
+   * @throws Error if trying to modify a non-malleable output or if subtracting more than the current amount
+   */
+  subtractAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    const subtractAmount = new BigNumber(amountSats);
+    if (subtractAmount.isGreaterThan(this._amountSats)) {
+      throw new Error("Cannot subtract more than the current amount");
+    }
+    this._amountSats = this._amountSats.minus(subtractAmount);
```

Comment:
Maybe cleaner to use set? 
```suggestion
    this.setAmount(this._amountSats.minus(subtractAmount));
```

Reply:
thanks !! [done](https://github.com/caravan-bitcoin/caravan/pull/114/commits/372f9197eb1a321131f3a18e1639f6d14f321e56#diff-db5fc022ab08b92fb2a606342674432f918748c762c9fcb907bc43d3531d512bR273)